src/request_handler.rs

```diff
@@ -87,6 +88,16 @@ impl RequestHandler {
     Ok(Response::new(Body::from(body.into_string())))
   }
 
+  async fn lookup_directory(path: &Path) -> io::Result<Vec<OsString>> {
```

Comment:
I'm not sure how I feel about `lookup_directory`, it don't sound to me like it returns the contents of the directory.

```suggestion
  async fn entry_file_names(path: &Path) -> io::Result<Vec<OsString>> {
```

Reply:
Hmm, not sure about `entry_file_names`. Does the `entry` refer to the directory? But I also don't like `lookup_directory`. Maybe `read_dir`? Or `read_dir_sorted`?